### PR TITLE
Json summaries

### PR DESCRIPTION
**Pull request by karel1980**
New and improved summaries

Needs a bit of cleaning up, but the summaries are nicer (and plentier, no 4245 documents summarized both in nl and fr)

### CODE REVIEW

File: transparentdemocracy/documents/summarize.py [hunk 1]
```diff
@@ -18,72 +22,33 @@
...
-
-
-class DocumentSummarizer():
-    def __init__(self, language="nl"):
+OLLAMA_MODEL = "llama3"
+PROMPT_STUFF = """Summarize the text in Dutch and in French. Here is the text:
+
+{text}
+
+Answer with a single json object. The dutch summary should be in a property called nl and the french summary should be in a property called fr."""
```
Comment (sandervh14) [inline transparentdemocracy/documents/summarize.py:34]: Does this consistently work?
I've found in json outputting instructions for an LLM at work, the LLM sometimes does not return proper JSON if you write out in "vage human terms" what to generate.
If so: we make it very explicit, like:

```
Return a JSON object, containing a Dutch summary (let's refer to this with the variable [DUTCH_SUMMARY]) and French summary (let's refer to this as [FRENCH_SUMMARY]) of the following text:

{text}

The returned JSON object must look like this:

{
  "nl": "[DUTCH_SUMMARY]",
  "fr": "[FRENCH_SUMMARY]"
}

The summary must not be longer than 5 sentences.
It should be written in layman terms, rather than using very judicial or political vocabulary.
```

And even an in-context example if you still find the LLM makes mistakes sometimes:

```
For example, this text:

Dit wetsontwerp reguleert de privé-opsporing van minderjarigen en autoriseert specifiek een privédetective om met toestemming van de ouders een kind jonger dan 12 jaar te ondervragen. De auteur vindt dat dit te ver gaat en kan leiden tot instrumentalisering door ouders. Daarom wordt voorgesteld deze bevoegdheid weg te laten en het getuigenis van minderjarigen alleen mogelijk te maken binnen het kader van een gerechtelijke procedure met alle daaraan verbonden garanties.

Must result in a JSON of this structure:

{
  "nl": "Laat toe om kinderen jonger dan 12 jaar te laten ondervragen door privé-detectives, indien de ouders van het kind toestemmen.",
  "fr": "Permettez une interrogation d'un enfant de moins de 12 ans par un détective privé, si les parents de l'enfant le permettent."
}
```

File: transparentdemocracy/documents/summarize.py [hunk 2]
```diff
@@ -250,19 +260,102 @@ def to_summary(path):
...
+        with open(path, 'r') as fp:
+            lines = fp.readlines()
+
+        lines = lines[2:]
+        with open(path, 'w') as fp:
+            fp.writelines(lines)
+
+    else:
+        raise Exception(f"Unknown action {action}")
+
```
Comment (sandervh14) [inline transparentdemocracy/documents/summarize.py:358]: I saw you wrote you would "improve the toolchain". I'll merge this PR so we get the up to date document summaries.
You can make the toolchain changes in a new PR, or on main...

**Comment by karel1980** [comment]
Omg sorry I made you review all that. The known actions stuff was just a bit of throwaway code to help me sort out the junk that the LLM was putting out.
I'll remove the known actions rubbish and try out your prompt improvements.
